pandas/core/frame.py

```diff
@@ -1570,16 +1570,44 @@ def from_csv(cls, path, header=0, sep=',', index_col=0, parse_dates=True,
 
     def to_sparse(self, fill_value=None, kind='block'):
         """
-        Convert to SparseDataFrame
+        Convert to SparseDataFrame.
+
+        Implement the sparse version of the DataFrame meaning that any data
+        matching a specific value it's omitted in the representation.
+        The sparse DataFrame allows for a more efficient storage.
 
         Parameters
         ----------
         fill_value : float, default NaN
+            The specific value that should be omitted in the representation.
         kind : {'block', 'integer'}
+            The kind of the SparseIndex tracking where data is not equal to
+            the fill value:
+
+            - 'block' tracks only the locations and sizes of blocks of data;
+            - 'integer' keeps an array with all the locations of the data.
+
+            The kind 'block' is recommended since it's more memory efficient.
 
         Returns
         -------
         y : SparseDataFrame
+
+        See Also
+        --------
+        DataFrame.to_dense :
+            converts the DataFrame back to the its dense form
+
+        Examples
+        --------
+
+        Compressing on the zero value.
+
+        >>> df = pd.DataFrame(np.random.randn(1000, 4))
+        >>> df.iloc[:995] = 0.
+        >>> sdf = df.to_sparse(fill_value=0.)
+        >>> sdf.density
+        0.005
```

Comment:
Nice example. It's just an opinion, feel free to leave it like this, if you think that being somehow more realistic is better, but I'd have something like: `pd.DataFrame([np.nan, np.nan, 1., np.nan])`. You can make it sparse with default arguments, then create another example with zeros instead of `NaN` and use `fill_value`. And if you find a nice way to illustrate the difference, I'd add an example with `kind='integer'`.